py-polars/src/lazyframe/visitor/expr_nodes.rs

```diff
@@ -571,8 +571,80 @@ pub(crate) fn into_py(py: Python<'_>, expr: &AExpr) -> PyResult<PyObject> {
                 FunctionExpr::ListExpr(_) => {
                     return Err(PyNotImplementedError::new_err("list expr"))
                 },
-                FunctionExpr::StringExpr(_) => {
-                    return Err(PyNotImplementedError::new_err("string expr"))
+                FunctionExpr::StringExpr(strfun) => match strfun {
+                    StringFunction::ConcatHorizontal {
+                        delimiter,
+                        ignore_nulls,
+                    } => ("concathorizontal", delimiter, ignore_nulls).to_object(py),
+                    StringFunction::ConcatVertical {
+                        delimiter,
+                        ignore_nulls,
+                    } => ("concatvertical", delimiter, ignore_nulls).to_object(py),
+                    StringFunction::Contains { literal, strict } => {
+                        ("contains", literal, strict).to_object(py)
+                    },
+                    StringFunction::CountMatches(_) => ("countmatches",).to_object(py),
+                    StringFunction::EndsWith => ("endswith",).to_object(py),
+                    StringFunction::Explode => ("explode",).to_object(py),
+                    StringFunction::Extract(_) => ("extract",).to_object(py),
+                    StringFunction::ExtractAll => ("extractall",).to_object(py),
+                    StringFunction::ExtractGroups { dtype, pat } => {
+                        ("extractgroups", Wrap(dtype.clone()).to_object(py), pat).to_object(py)
+                    },
+                    StringFunction::Find { literal, strict } => {
+                        ("find", literal, strict).to_object(py)
+                    },
+                    StringFunction::ToInteger(_) => ("tointeger",).to_object(py),
+                    StringFunction::LenBytes => ("lenbytes",).to_object(py),
+                    StringFunction::LenChars => ("lenchars",).to_object(py),
+                    StringFunction::Lowercase => ("lowercase",).to_object(py),
+                    StringFunction::JsonDecode {
+                        dtype,
+                        infer_schema_len,
+                    } => (
+                        "jsondecode",
+                        Wrap(dtype.as_ref().unwrap().clone()).to_object(py),
```

Comment:
question (for @ritchie46): are these `dtypes` that live inside the variants different from the ones that `AExpr.to_field(...)` would return?

Reply:
You should always trust `AExpr::to_field`. That this variant holds a `dtype` is an implementation detail. Those details are now leaked and might be changed. So be aware of that.

Reply:
We could therefore just not leak these out: it makes the mapping not one-to-one, but if `to_field` is the correct source of truth then it makes sense to not lead someone down a bad path by providing these slots in the first place.

Reply:
Yes, agree.